parser/src/main/java/me/vertretungsplan/parser/IphisParser.java

```diff
@@ -0,0 +1,358 @@
+            }
+        }
+        // Link teacher IDs to their names
+        HashMap<String, String> teachersHashMap = null;
+        if (teachers != null) {
+            teachersHashMap = new HashMap<>();
+            for (int i = 0; i < teachers.length(); i++) {
+                JSONObject teacher = teachers.getJSONObject(i);
+                teachersHashMap.put(teacher.getString("id"), teacher.getString("name"));
+            }
+        }
+        // Add changes to SubstitutionSchedule
+        LocalDate currentDate = LocalDate.now();
+        SubstitutionScheduleDay substitutionScheduleDay = new SubstitutionScheduleDay();
+        substitutionScheduleDay.setDate(currentDate);
+        for (int i = 0; i < changes.length(); i++) {
+            final JSONObject change = changes.getJSONObject(i);
+            final Substitution substitution = getSubstitution(change, coursesHashMap, teachersHashMap);
+            final LocalDate substitutionDate = new LocalDate(change.getString("datum"));
+
+            // If starting date of change does not equal date of SubstitutionScheduleDay
+            if (!substitutionDate.isEqual(currentDate)) {
+                if (!substitutionScheduleDay.getSubstitutions().isEmpty()) {
+                    substitutionSchedule.addDay(substitutionScheduleDay);
+                }
+                substitutionScheduleDay = new SubstitutionScheduleDay();
+                substitutionScheduleDay.setDate(substitutionDate);
+                currentDate = substitutionDate;
+            }
+            substitutionScheduleDay.addSubstitution(substitution);
+        }
+        substitutionSchedule.addDay(substitutionScheduleDay);
+    }
+
+    private String[] getSQLArray(String data) {
+        String[] retArray = {};
+        Pattern pattern = Pattern.compile("\\{(.*?)}");
+        Matcher matcher = pattern.matcher(data);
+        if (matcher.find()) {
+            retArray = matcher.group(1).split(",");
+        }
+        return retArray;
+    }
+
+    private Substitution getSubstitution(JSONObject change, HashMap<String, String> gradesHashMap,
+                                         HashMap<String, String> teachersHashMap)
+            throws IOException, JSONException {
+        final Substitution substitution = new Substitution();
+        // Set class(es)
+        final String[] classIds = getSQLArray(change.getString("id_klasse"));
+        if (classIds.length > 0) {
+            if (gradesHashMap == null) {
+                throw new IOException("Change references a grade but grades are empty.");
+            }
+            final HashSet<String> classes = new HashSet<>();
+            for (String classId : classIds) {
+                classes.add(gradesHashMap.get(classId));
+            }
+            substitution.setClasses(classes);
+        }
+        // Set type
+        final String type = change.getString("aenderungsgrund");
+        substitution.setType(type);
```

Comment:
Wenn der Änderungsgrund wie in den Beispieldaten immer mit einem Leerzeichen endet, sollte das hier (z.B. über `.trim()`) entfernt werden.